Mathlib/RingTheory/Localization/Basic.lean

```diff
@@ -786,8 +786,22 @@ noncomputable def atUnits (H : M ≤ IsUnit.submonoid R) : R ≃ₐ[R] S := by
 
 end at_units
 
+
 end IsLocalization
 
+
+section
+
+variable [IsLocalization M S] (M) (Q : Type*) [CommSemiring Q] {g : R →+* P} [Algebra P Q]
+
+theorem map_injective_of_injective (hg : Function.Injective g)
+    [i: IsLocalization (M.map g : Submonoid P) Q] :
```

Comment:
The variable i is not used in the proof. But if I take it away, then theorem in line 1433 breaks.

Reply:
The proof there should be fixed by something like `haveI : IsLocalization (M.map (algebraMap R S)) Sₘ := i` instead.

Reply:
Or maybe we should make `Algebra.algebraMapSubmonoid` reducible.

Reply:
I changed the proof in line 1433 so that now it works.